### PR TITLE
Redis configuration extraction

### PR DESCRIPTION
**Pull request by juanignaciosl**
Part of #8439

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@zenitraM , @javitonino , could you both review this, please? There are a few subtle differences between the way that Resque instantiated the Redis connection vs the way the editor did that I'm not _that_ sure about:

- Resque didn't pick `driver` from the configuration ([r](https://github.com/CartoDB/cartodb/pull/8535/files#diff-22ab337d79a7ad5c4588294a470c510aL10) vs [e](https://github.com/CartoDB/cartodb/pull/8535/files#diff-07c913c00bb08eb60ce2c64b4ef8fbdeL17)).
- Only the Editor used `ENV['REDIS_PORT']` if present ([e](https://github.com/CartoDB/cartodb/pull/8535/files#diff-07c913c00bb08eb60ce2c64b4ef8fbdeL22)).
- Resque didn't pick the database ([r](https://github.com/CartoDB/cartodb/pull/8535/files#diff-22ab337d79a7ad5c4588294a470c510aL18) vs [e](https://github.com/CartoDB/cartodb/pull/8535/files#diff-07c913c00bb08eb60ce2c64b4ef8fbdeL46)).

**Comment by javitonino** [comment]
> Resque didn't pick driver from the configuration (r vs e).
I think it probably should, it is used to force hiredis.

> Only the Editor used ENV['REDIS_PORT'] if present (e).
This is used for parallel testing, and the queues are not run there, so it should be alright.

> Resque didn't pick the database (r vs e).
That seems reasonable, as resque uses its own database.

File: lib/redis_factory.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    Redis.new(configuration)
+  end
+
+  def self.get_conf
+    # TODO: because of #8439 we're testing different Redis timeouts for connection, R and W.
+    # After getting a final solution timeout configuration should be read from the file again
+    redis_conf = conf.select { |k, _| [:host, :port, :driver, :tcp_keepalive].include?(k) }
+    redis_conf[:connect_timeout] = 2
+    redis_conf[:read_timeout] = 3
+    redis_conf[:write_timeout] = 5
```
Comment (javitonino) [inline lib/redis_factory.rb]: I would have started reading from the config file in this PR, and left these values as defaults. But up to you if you prefer to split it.

Comment (juanignaciosl) [inline lib/redis_factory.rb]: Ok, I will. I just wanted to do the extraction before and ask for a first review.

File: lib/redis_factory.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    redis_migrator_logs: 6
+  }.freeze
+
+  def self.conf
+    # Note that `symbolize_keys` creates a new instance, which is relevant to avoid side effects
+    Cartodb.config[:redis].symbolize_keys
+  end
+
+  private_class_method :get_conf
+  private_class_method :databases
```
Comment (javitonino) [inline lib/redis_factory.rb]: `conf` should probably be private too.

**Comment by javitonino** [comment]
This looks good, thanks for the refactor. Left a couple of minor comments, but nothing of notice. I see the same differences you pointed out, but I think they are not problematic. So :+1: for me.

File: lib/redis_factory.rb (hunk 2 above)
Comment (juanignaciosl) [inline lib/redis_factory.rb]: Yep!

**Comment by juanignaciosl** [comment]
@javitonino CR this, please. Before merging I'll change the configuration adding the new timeouts without removing the global one.

**Comment by javitonino** [comment]
LGTM :+1:

**Pull request merged**